src/Backend/Snowflake/Importer.php

```diff
@@ -118,36 +119,36 @@ private function runQuery(string $query, ?string $timerName = null): void
         }
     }
 
+    /**
+     * @param Storage\Snowflake\Table $destination
+     */
     private function importToStagingTable(
         Storage\SourceInterface $source,
         Storage\DestinationInterface $destination,
         ImportOptions $importOptions
     ): void {
         $adapter = $source->getBackendImportAdapter($this);
-        if (!$adapter instanceof SnowflakeImportAdapterInterface) {
-            throw new \Exception(sprintf(
-                'Adapter "%s" must implement "SnowflakeImportAdapterInterface".',
-                get_class($adapter)
-            ));
-        }
         $commands = $adapter->getCopyCommands($destination, $importOptions, $this->importState->getStagingTableName());
         try {
-            $this->importState->addImportedRowsCount(
-                $adapter->executeCopyCommands(
-                    $commands,
-                    $this->connection,
-                    $destination,
-                    $importOptions,
-                    $this->importState
-                )
-            );
+            $this->importState->startTimer('copyToStaging');
+            foreach ($commands as $command) {
+                $this->connection->query($command);
+            }
+            $this->importState->stopTimer('copyToStaging');
         } catch (\Throwable $e) {
             $stringCode = Exception::INVALID_SOURCE_DATA;
             if (strpos($e->getMessage(), 'was not found') !== false) {
                 $stringCode = Exception::MANDATORY_FILE_NOT_FOUND;
             }
             throw new Exception('Load error: ' . $e->getMessage(), $stringCode, $e);
         }
+
+        $rows = $this->connection->fetchAll(sprintf(
```

Comment:
tu count query by taky mohl poskytnout ten sqlBuilder at se to nemichaji urovne abstrakce